### ISSUE TITLE
Kafka broker IDs get different values if "epicli apply ..." is run repeatedly

### ISSUE THREAD

**Issue by apopaa** [epiphany-platform/epiphany#1344]
This is the code in epicli that sets the broker.id (taken from [server.properties.j2](https://github.com/epiphany-platform/epiphany/blob/develop/core/src/epicli/data/common/ansible/playbooks/roles/kafka/templates/server.properties.j2)  )
```
# The id of the broker. This must be set to a unique integer for each broker.
{% for url in kafka_hosts.split(',') %}
  {%- set url_host = url.split(':')[0] -%}
  {%- if url_host == ansible_fqdn or url_host in ansible_all_ipv4_addresses
         or url_host == ansible_ssh_host -%}
broker.id={{ loop.index0 }}
```
And this is the code that constructs that kafka_hosts string (taken from [kafka.yaml](https://github.com/epiphany-platform/epiphany/blob/develop/core/src/epicli/data/common/ansible/playbooks/group_vars/kafka.yml) ) :
```
kafka_hosts: "{{ groups['kafka']|join(':9092,') }}:9092"
```
According to [this bug report](https://github.com/confluentinc/cp-ansible/issues/77) , because groups['kafka'] is a dictionary, which is not ordered in Python, you're not guaranteed to get the same order every time you loop over it and thus to get the same position for a Kafka server in the iteration. But we're assigning broker.id based on position in the iteration, which means if you're unlucky you get one position and one broker.id the first time you apply and another position and another broker.id the second time you apply.
I'm not sure if the inconsistency happens when the string is constructed or when it is split, but I think both can be improved by just adding a "sort" filter, which would guarantee an order.

Epiphany 0.6.0 with the officially released epipcli Docker container was used against Azure infrastructure.

**Comment by sk4zuzu** [comment]
Hi @apopaa, thank you for reporting the problem!

I don't think this is entirely correct interpretation of the issue. Please look at this comment https://github.com/ansible/ansible/issues/34861#issuecomment-638188252 and the discussion above it. Could you provide more info about which python, ansbile and epiphany versions are you using to reproduce this issue? Also which provider type are you using (AWS, Azure or any)?

**Comment by apopaa** [comment]
Added more info in bug description.

**Comment by mkyc** [comment]
@przemyslavic please have look at this in current version

**Comment by przemyslavic** [comment]
Fixed in #2356 by recreating `meta.properties`.
